src/screen/dashboard.rs

```diff
@@ -128,105 +128,177 @@ impl Dashboard {
+                                                    .get_server_chathistory_message_reference_type(
+                                                        server,
+                                                    );
+
+                                                let oldest_message_reference = self
+                                                    .get_oldest_message_reference(
+                                                        server,
+                                                        &target,
+                                                        message_reference_type,
+                                                    );
+
+                                                clients.send_chathistory_request(
+                                                    ChatHistorySubcommand::Before,
+                                                    server,
+                                                    &target,
+                                                    oldest_message_reference,
+                                                );
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            Some(buffer::Event::ChatHistoryBeforeRequest) => {
+                                if let Some(buffer) = pane.buffer.data() {
+                                    let server = buffer.server();
+
+                                    if clients.get_server_supports_chathistory(server) {
+                                        if let Some(target) = buffer.target() {
+                                            let message_reference_type = clients
+                                                .get_server_chathistory_message_reference_type(
+                                                    server,
+                                                );
+
+                                            let oldest_message_reference = self
+                                                .get_oldest_message_reference(
+                                                    server,
+                                                    &target,
+                                                    message_reference_type,
+                                                );
+
+                                            clients.send_chathistory_request(
+                                                ChatHistorySubcommand::Before,
+                                                server,
+                                                &target,
+                                                oldest_message_reference,
+                                            );
+                                        }
+                                    }
+                                }
+                            }
+                            _ => (),
```

Comment:
`_` is dangerous as the compiler won't notify us when new variants are added. Match on `None` instead to fully exhaust the match

Reply:
Hadn't thought about it that way before, thanks for the suggestion/tip :)